### DISCUSSION TITLE
This app isn't available for your device because it was made for an older version of Android

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#2890] [Q&A]
On Samsung phone (One UI, Knox 3.9, Android 13) I'm getting "This app isn't available for your device because it was made for an older version of Android" on Google Play. Same app installs normally on Xiaomi phone (MIUI 14, Android 13).

Does anybody knows why is this so?

I'm totally confused with all these Android versions and restrictions and security updates and...

**Comment by FliplineStudios** [comment]
Is this an older app, or something that you've published recently?  And do you have info on the AIR SDK you used, the build-tools you used, and the targetSdkVersion in the app?

Google has a rolling requirement for target SDK versions now, so at the moment everything needs to target at least API Level 33 (including apps that have been live and untouched in the Play Store for years).  And with newer AIR SDKs, you would also need to be using Android build-tools for 33 set up in your platformsdk path.  I'm pretty sure Google shows that "isn't available" message when you're targeting something too old, though in some cases I've heard it can mistakenly come up if the device fails to meet other requirements.

In case you haven't set it up already, I believe the steps you would need to take to ensure you're targeting 33 or later would be:

- make sure you're targeting 33 in your application XML (android:targetSdkVersion="33" in your uses-sdk tag)
- download Android sdk build-tools for 33 (I'm using 33.0.2)
- point to your Android SDK directory (with adt -platformsdk flag, or with adt.cfg, etc.)

The AIR SDK Dev site has more info here if needed:
https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles

If you're already targeting 33, please share some more details though!  I can compare to some app updates we've recently done which in theory aren't showing that message as far as I can tell...

**Reply by MalacTheLittle** [reply]
It's an old app, last update was made in 2021 and targeted API level is 29 (AIR SDK 33.1, probably AIR33.1.1.633).
Confusing part is why some phones (like Samsung) doesn't allow install of older apps, and some does (like Xiaomi). Personally, my Xiaomi installs any older app. Also, if I run app that targets API level 25 and that app uses Documents folder as storage, on Xiaomi it works like there's no "Android 11 Documents folder policy restrictions".

The thing is that I have some older apps that uses Document folder as storage. Now, if I want to make an update to API 33 I have to change apps storage path to ApplicationStorageFolder. The thing that concerning me is that people with Xiaomi like mobiles will get an update and they'll have to download all the app assets again to ApplicationStorageFolder just because I have to update app according to Google policy.  

There's an option to keep old apk's in new release, so I can target my new update as minVersion=30 and targetVersion=33. That way older phones that runs Android 10 won't get update which is fine. But, what will happen with newer phones that runs Android 13 and has installed older version of app targeting API level 29? Will they get update? I want to avoid that so users doesn't need to download app assets again.

**Reply by FliplineStudios** [reply]
Hmm does the Xiaomi phone use Google Play, or a different app store?  I think the limitation is from the Google Play Store itself which is how Google enforces their rule changes.

> But, what will happen with newer phones that runs Android 13 and has installed older version of app targeting API level 29? Will they get update?

Yeah, as long as their device's Android version is >= the minSdkVersion, it will (likely automatically) update to the new app version.  

It's a huge headache with all of these API changes and requirement changes from Google, especially since they won't let older apps just remain as-is anymore with these forced updates...

**Reply by MalacTheLittle** [reply]
Yes, Xiaomi use Google Play which means that Google at the end doesn't control situation with all the new rules and restrictions.
It's completely confusing and I'm a bit lost in my options and decisions what to do.

**Reply by FliplineStudios** [reply]
Here's more of the details about Google's changes, where existing apps need to keep being updated to remain available in the Play Store (in the App Availability Requirements section near the bottom):
https://support.google.com/googleplay/android-developer/answer/11926878?hl=en

It's a bit surprising that your Xiaomi can still install it from the Play Store... though they do mention "will not be available to **new** users", so maybe if you had installed it on that device before these new requirements kicked in, they'll keep letting you install it on that device for now?  Google is never super clear about these things.  But users who never had it on their device before likely can't install it at all...

Have you seen any policy warnings in the Google Play Console (in the Inbox or Policy Status) for that app?  They've sent us warnings in there and emails as well once we were getting close to a deadline of apps needing their annual update, mentioning that new users wouldn't be able to find it or download it if we didn't update.

**Reply by MalacTheLittle** [reply]
Yes, I got those policy warnings. I finally caught some time to work on that, but I can't find any clear answers on the issues I'm facing with.

> though they do mention "will not be available to **new** users", so maybe if you had installed it on that device before these new requirements kicked in, they'll keep letting you install it on that device for now?

You may be right with this one... But I got my Xiaomi half a year ago and I installed app targeting level 25 which is using Documents folder as storage and it works without a problem. Does that mean that Documents folder restriction also doesn't apply to users that had apps installed before Android 11 restrictions or is it inconsistency between manufacturers?

**Comment by FliplineStudios** [comment]
> Does that mean that Documents folder restriction also doesn't apply to users that had apps installed before Android 11 restrictions or is it inconsistency between manufacturers?

That's a great question, I'm not really sure how the changes/permissions are handled for existing users, we haven't used the Documents folder ourselves...  hopefully someone else out there has some experience with this!

**Comment by MalacTheLittle** [comment]
I made a test with one of my apps. I made new release with minVersion +1 of old apk targetVersion and have included old apk builds in release.

I can confirm that devices with installed older (lower) targetVersion API level apps than device Android API level **will** get an app update if minVersion of app is smaller or equal to devices Android API level. The result is expected, but was hoping keeping old apk/aab builds will make a difference :/

Now, the main question remains... Why there's a difference between manufacturers old Android version apps handling?
